tests/test_resolver.py

```diff
@@ -10,9 +10,33 @@
 
 # This environment var must be set before the usd imports.
 os.environ["TF_DEBUG"] = "OPENASSETIO_RESOLVER"
-from pxr import Plug, Usd
+from pxr import Plug, Usd, Ar
 
 
+# Assume OpenAssetIO is configured as the custom primary resolver for
+# all tests. If you're wondering where this is configured, it may
+# just be set via the `PXR_PLUGINPATH_NAME` environment variable.
+
+# Verify OpenAssetIO configured as the AR resolver.
+@pytest.fixture(autouse=True)
+def openassetio_configured():
```

Comment:
[consider][minor] We do try to put utility functions at the bottom of the file, with the general principle that we should be able to read source code from high level to low level as we scroll down.

This style has been our preference so far (though not always obeyed).  Happy to discuss if you have a counter opinion.

Reply:
Aye i'm not doing this intentionally, just habit, the idea being each unit of code maintains maximum local cohesion, so you can almost treat sections of the file as seperate, leaving yourself open to easily factor out to real boundaries (ie, new files) later when the scope of the sections justify it.

Will conform to standards.

Reply:
To me there is an argument that utilities should be non-local as they may well be shared, and that makes it clear. ie, if you had a large file of tests, you could accidentally re-implement something several time as you hadn't seen it, as it was local to some specific use?

Reply:
Aye, very fair and true. A true module _extremist_ might argue that duplicate utility implementations is fine, and even preferred, in order to keep cross-module dependencies mapped exclusively to data communication, and not to implementational detail.

There's a section in `Clean Code` about this, page 140 if you're interested, although it speaks exclusively about module locality, my extension of the reasoning to file scope is personal and tenuous, and not something I'd wanna stand by :P.

Reply:
You know, reading that section of the book again, I find it hard to imagine how I went from there to, "Things should be really close together in the source file,) you could easily take a fully contrary interpretation. Ach well, no matter.

Reply:
https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/13/commits/b22d39f05c43d3a7021a51aafef71d110acad39d

Util funcs at the bottom